l10n_br_hr_payroll_connector/models/hr_payslip_run.py

```diff
@@ -0,0 +1,34 @@
+from openerp import models, api
+from openerp.addons.l10n_br_hr_payroll.models.profiling import (
+    clear_prof_data,
+    log_prof_data
+)
+import logging
+
+from openerp.addons.connector.queue.job import job
+from openerp.addons.connector.session import ConnectorSession
+
+_logger = logging.getLogger(__name__)
+
+
+class HrPayslipRun(models.Model):
+    _inherit = "hr.payslip.run"
+
+    @api.multi
+    def gerar_holerites(self):
+        session = ConnectorSession(
+            self.env.cr, self.env.uid, self.env.context)
+        for contrato in self.contract_id:
+            job_gerar_holerite.delay(session, self.id, contrato.id)
+        self.verificar_holerites_gerados()
```

Comment:
@gabrielcardoso21 e @mileo acredito que as duas funções depois desse for vão ser acionadas antes de ter terminados de processar todos os holerites perdendo assim sua real função, uma vez que o for vai ser executado de forma 'rapida'.

Reply:
@lfdivino Fiz um commit de correção disso. Coloquei para executar essas funções toda vez que for gerado um novo holerite.

Reply:
Mas fazendo desta forma você está aumentado de novo o tempo de execução, provavelmente a melhor for de fazer seria verificar se existe alguma função no módulo de job.queue onde ele aguarde a execução de toda a fila e só depois executar a outra função, ou algo semelhante a isso.

Reply:
Pesquisamos ontem e não existe esta função, seria necessário outra refatoração.

Reply:
@lfdivino Coloquei uma verficação no próprio job para executar essas funções somente se não houver job pendente, ou seja, apenas no último job a ser executado.